nox/manifest.py

```diff
@@ -138,25 +154,43 @@ def make_session(self, name, func):
             Sequence[~nox.session.Session]: A sequence of Session objects
                 bound to this manifest and configuration.
         """
+        sessions = []
+
+        # If the func has the python attribute set to a list, we'll need to
+        # expand them.
+        if isinstance(func.python, (list, tuple, set)):
+            for python in func.python:
+                single_func = _copy_func(func)
+                single_func.python = python
+                sessions.extend(self.make_session(name, single_func))
+
+            return sessions
+
         # Simple case: If this function is not parametrized, then make
         # a simple session
         if not hasattr(func, 'parametrize'):
-            session = Session(name, None, func, self._config, self)
-            return (session,)
+            if func.python is not None:
+                long_name = '{}-{}'.format(name, func.python)
+            else:
+                long_name = name
+            session = SessionRunner(name, long_name, func, self._config, self)
```

Comment:
Could you add the keywords here, even if it forces line breaks? There are enough arguments that it would be easier to read.

Reply:
Will do.